test/language/expressions/class/super-evaluation-order.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2020 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: prod-SuperCall
+description: SuperCall should evaluate Arguments prior to checking IsConstructable
+info: |
+  GetSuperConstructor ( )
+
+  [...]
+  1. Let _func_ be ! GetSuperConstructor().
+  2. Let _argList_ be ? ArgumentListEvaluation of |Arguments|.
+  3. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
+  [...]
+features: [default-parameters]
+---*/
+
+var i;
+class A extends function () { throw new TypeError(); } {
```

Comment:
@ljharb 

```js
var x = 0;
class C extends Object {
  constructor() {
    super(x = 1);
  }
}

Object.setPrototypeOf(C, parseInt);
```

this will get you good for the assertions.